File: server-node/src/routers/jsonConv.js
const express = require('express');
const router = express.Router();
const multer = require('multer');
const upload = multer({ storage: multer.memoryStorage() });
const { unifiedBatchHandler, readData } = require('../utils/helpers');

async function processToJson(buffer, { orient, indent, sheet_name, isCsv, filename }) {
    const data = readData(buffer, filename, sheet_name);
    const content = JSON.stringify(data, null, parseInt(indent) || 4);
    return { buffer: Buffer.from(content), extension: '.json' };
}

router.post('/convert-to-json', upload.array('files'), async (req, res) => {
    const { orient, indent, sheet_name, all_sheets } = req.body;
    await unifiedBatchHandler(req, res, processToJson,
        { orient, indent, sheet_name, all_sheets },
        "JSON Conversion", "");
});

module.exports = router;

File: backend/routers/json_conv.py
from typing import List
from fastapi import APIRouter, UploadFile, File, Form, Depends, BackgroundTasks
from backend.core.auth import get_current_user
from backend.utils.helpers import unified_batch_handler
from tools.json_converter import convert_to_json

router = APIRouter(prefix="/api/file", tags=["json"])

@router.post("/convert-to-json")
async def convert_to_json_api(
    files: List[UploadFile] = File(...),
    orient: str = Form("records"),
    indent: int = Form(4),
    sheet_name: str = Form(None),
    all_sheets: bool = Form(False),
    user=Depends(get_current_user),
    background_tasks: BackgroundTasks = BackgroundTasks()
):
    return await unified_batch_handler(
        files,
        convert_to_json,
        {"orient": orient, "indent": indent, "sheet_name": sheet_name, "apply_all_sheets": all_sheets},
        "JSON Conversion",
        "", # extension handled by processor
        user=user,
        background_tasks=background_tasks
    )
